feat: Reorganize password onboarding

ptbrowne opened the pull request:
- Use a Stepper to show onboarding progress
- Check if vault has been configured to bypass first steps
- Check hint to show a message if already configured
- Show warning in stepper if vault configured but hint is not configured

ptbrowne committed:
chore: Update cozy-client

ptbrowne committed:
feat: Reorganize password onboarding

- Use a Stepper to show onboarding progress
- Check if vault has been configured to bypass first steps
- Check hint to show a message if already configured
- Show warning in stepper if vault configured but hint is not configured

Crash-- reviewed (commented):
Can you fix the tests and ask again for review?

ptbrowne committed:
test: Repair tests

- Move configure extension tests to their own file
- Tests that sidebar is not rendered when vault is not configured

ptbrowne committed:
refactor: Extract component from InstallationPage

Will be easier for tests

ptbrowne committed:
refactor: Remove disabled tests

Those tests will be reintroduced in another form in a later commit

Crash-- commented on src/components/HintStep.jsx:
```diff
@@ -13,14 +13,13 @@ import Spinner from 'cozy-ui/transpiled/react/Spinner'
-const DumbHintPage = props => {
-  const { client, history } = props
+const DumbHintStep = props => {
+  const { client } = props
```
```suggestion
  const { client, goToNextStep } = props
```

Crash-- commented on src/components/HintStep.jsx:
```diff
@@ -13,14 +13,13 @@ import Spinner from 'cozy-ui/transpiled/react/Spinner'
-  const [loading, setLoading] = useState(true)
 
-  const goToNextStep = () => history.push('/installation')
+  const goToNextStep = () => props.goToNextStep()
```
```suggestion
```

Crash-- commented on src/components/InstallationPage.jsx:
```diff
@@ -0,0 +1,112 @@
+  }, [activeStep])
+
+  return (
+    <div style={{ maxWidth: 800, margin: '0 auto', width: '100%' }}>
```
inline css ? 🤔

Crash-- commented on src/hint.js:
```diff
@@ -0,0 +1,18 @@
+    const hint = await client
+      .getStackClient()
+      .collection('io.cozy.settings')
+      .get('hint')
```
can't you : 
```
client.query(client.get('io.cozy.settings', 'hint'))
``` 
?

ptbrowne replied:
No this is not possible as the hint does not return any data (204 No content), and it does not work with `query` in this case. Thank for asking though, I had not checked.

ptbrowne replied:
https://github.com/cozy/cozy-passwords/pull/25/commits/229f29e3604aa4e2d50396fc2176d63cc69eb78e makes it clearer.

Crash-- commented on src/components/App.jsx:
```diff
@@ -93,24 +40,32 @@ export const DumbApp = () => {
-        </HashRouter>
-      </MuiCozyTheme>
-    </BreakpointsProvider>
+    <BitwardenSettingsContext.Provider value={queryResult.data}>
```
I like this idea 👍

ptbrowne committed:
refactor: No need for intermediate function

ptbrowne committed:
refactor: No inline styles

ptbrowne committed:
refactor: Make it clear than we cannot fetch the hint, only know if it
exists
